lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -832,6 +832,30 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
     end
   end
 
+  @spec process_eth1_data(BeaconState.t(), BeaconBlockBody.t()) ::
+          {:ok, BeaconState.t()} | {:error, binary}
+  def process_eth1_data(
+        %BeaconState{} = state,
+        %BeaconBlockBody{eth1_data: eth1_data} = _body
+      ) do
+    updated_state =
+      %BeaconState{
+        state
+        | eth1_data_votes: List.insert_at(state.eth1_data_votes, -1, eth1_data)
+      }
+
+    if Enum.count(updated_state.eth1_data_votes, &(&1 == eth1_data)) * 2 >
+         ChainSpec.get("EPOCHS_PER_ETH1_VOTING_PERIOD") * ChainSpec.get("SLOTS_PER_EPOCH") do
+      {:ok,
+       %BeaconState{
+         updated_state
+         | eth1_data: eth1_data
+       }}
+    else
+      {:ok, updated_state}
```

Comment:
nit: I would calculate `eth1_data_votes` and `eth1_data` and at the end return the updated value

```
{:ok, %BeaconState{
  state
  | eth1_data: eth1_data, 
    eth1_data_votes: eth1_data_votes}
  }
}
```